### PR TITLE
20% npm install speedup: cache extras by package.json string

### PR DESCRIPTION
**Pull request by mikesherov**
## TL:DR
~20% speed up in the average npm install run when combined with https://github.com/npm/hosted-git-info/pull/24

## Details
This package used to actually cache package.json lookups, until https://github.com/npm/read-package-json/commit/b746f651da96aedca33e1f0e08c305ed055c417d which was an attempt to fix https://github.com/npm/npm/issues/7074 Deleting the caching mechanism definitely helped fix a correctness issue, but threw the baby out with the bath water. This PR introduces a less aggressive cache by caching by the *true* cache key, which is the string contents of a package.json file. Should be 100% safe to cache results based on this... I believe.

This lets us speed up builds quite a bit!

### Before:
```
added 1764 packages in 51.943s
added 1764 packages in 51.135s
added 1764 packages in 50.688s
added 1764 packages in 52.048s
added 1764 packages in 52.156s
added 1764 packages in 54.817s
```
### After
```
added 1764 packages in 42.428s
added 1764 packages in 45.116s
added 1764 packages in 42.085s
```

Let me know if you have any questions

### CODE REVIEW

**Comment by iarna** [comment]
This was merged into v2.0.7 as 839b56e. Thank you so much! =)
